core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockReadHandlerTest.java

```diff
@@ -0,0 +1,311 @@
+  private static final PropertyKey.Template QUOTA_TEMPLATE =
+      PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA;
+  private static final PropertyKey.Template MEDIUM_TEMPLATE =
+      PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_MEDIUMTYPE;
+
+  // test configuration values
+  private static final String TIER_0_ALIAS = "ALIAS_0";
+  private static final String TIER_1_ALIAS = "ALIAS_1";
+  private static final String WORKER_DATA_FOLDER = "TestWorker";
+  private static final String WORKER_TMP_FOLDER = "TestTmpWorker";
+
+  @Rule
+  public ConfigurationRule mConfiguration = new ConfigurationRule(
+      new ImmutableMap.Builder<PropertyKey, Object>()
+          // disable paging for this test
+          .put(PropertyKey.USER_BLOCK_STORE_TYPE, BlockStoreType.FILE)
+          // use 2 tiers to test possible moves of blocks
+          .put(PropertyKey.WORKER_TIERED_STORE_LEVELS, 2)
+          // disable alignment so that no space needs to be reserved
+          // on directories
+          .put(PropertyKey.WORKER_MANAGEMENT_TIER_ALIGN_ENABLED, false)
+          .put(PropertyKey.WORKER_DATA_FOLDER, WORKER_DATA_FOLDER)
+          .put(PropertyKey.WORKER_DATA_TMP_FOLDER, WORKER_TMP_FOLDER)
+          // tier configurations
+          .put(ALIAS_TEMPLATE.format(0), TIER_0_ALIAS)
+          .put(ALIAS_TEMPLATE.format(1), TIER_1_ALIAS)
+          .put(QUOTA_TEMPLATE.format(0), "1024")
+          .put(QUOTA_TEMPLATE.format(1), "1024")
+          .put(MEDIUM_TEMPLATE.format(0), "MEM")
+          .put(MEDIUM_TEMPLATE.format(1), "SSD")
+          .build(),
+      Configuration.modifiableGlobal()
+  );
+
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  // each tier's storage directory
+  private File mTier0Dir;
+  private File mTier1Dir;
+
+  // mocked response observer to interact with the handler
+  private TestResponseObserver mResponseObserver;
+
+  private final Closer mCloser = Closer.create();
+  // local store that backs the short circuit read
+  private BlockStore mBlockStore;
+  private ShortCircuitBlockReadHandler mTestHandler;
+
+  @Before
+  public void before() throws Exception {
+    // set up storage tier directories
+    mTier0Dir = mFolder.newFolder();
+    mTier1Dir = mFolder.newFolder();
+    mConfiguration.set(PATH_TEMPLATE.format(0), mTier0Dir.getAbsolutePath());
+    mConfiguration.set(PATH_TEMPLATE.format(1), mTier1Dir.getAbsolutePath());
+
+    // set up response observer
+    mResponseObserver = new TestResponseObserver();
+
+    // set up local storage
+    // todo(yangchen): BlockStore should not worry about the logic of reporting to master,
+    // could leave that work to BlockWorker so that we can get rid of these irrelevant mocks
+    BlockMasterClientPool pool = mock(BlockMasterClientPool.class);
+    BlockMasterClient cli = mock(BlockMasterClient.class);
+    doReturn(cli).when(pool).createNewResource();
+    doReturn(cli).when(pool).acquire();
```

Comment:
nit: It's more idiomatic to write `when(pool.createNewResource()).thenReturn(cli)`.

Reply:
Done.